libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -347,7 +347,6 @@ bool IRGeneratorForStatements::visit(BinaryOperation const& _binOp)
 			"\n";
 	else if (TokenTraits::isCompareOp(op))
 	{
-		solUnimplementedAssert(commonType->category() != Type::Category::Function, "");
```

Comment:
Please add a check that we only allow `==` and `!=` for functions.

Reply:
Also, I think this allows comparison for internal and external functions. Comparison for external functions is more complicated because they use two variables. There should be a function to combine the variables into a single one, or you just compare both.